infrahub_sdk/client.py

```diff
@@ -727,14 +729,49 @@ async def filters(
                 prefetch_relationships=prefetch_relationships,
                 timeout=timeout,
             )
-            nodes.extend(process_result["nodes"])
-            related_nodes.extend(process_result["related_nodes"])
-
-            remaining_items = response[schema.kind].get("count", 0) - (page_offset + self.pagination_size)
-            if remaining_items < 0 or offset is not None or limit is not None:
-                has_remaining_items = False
-
-            page_number += 1
+            return response, process_result
+
+        async def process_batch(schema_kind: str) -> tuple[list[InfrahubNode], list[InfrahubNode]]:
+            """Process queries in batch mode."""
+            nodes = []
+            related_nodes = []
+            batch_process = await self.create_batch()
+            resp = await self.execute_graphql(query=f"query {{ {schema_kind} {{ count }} }}")
```

Comment:
Would be good to move this into a dedicated `count` method, there's already an issue to track that https://github.com/opsmill/infrahub-sdk-python/issues/158

Reply:
I have now moved it over to using `client.count()`.